modules/tide_edit_protection/tide_edit_protection.module

```diff
@@ -142,3 +144,11 @@ function tide_edit_protection_preprocess_status_messages(&$variables) {
     }
   }
 }
+
+/**
+ * Implements hook_entity_predelete().
+ */
+function tide_edit_protection_entity_predelete(EntityInterface $entity) {
+  \Drupal::service('content_lock')
```

Comment:
@vincent-gao imo we should have some comments indicating why we need this hook, probably referencing the Drupal issues.